tests/tracing/instrumentation/test_action_executor.py

```diff
@@ -1,17 +1,28 @@
 import pytest
 
-from typing import Any, Dict, Sequence, Text, Optional
+from typing import Any, Dict, Sequence, Text, Optional, List
 from unittest.mock import Mock
 from pytest import MonkeyPatch
 from opentelemetry.sdk.trace import ReadableSpan, TracerProvider
 from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
 from opentelemetry import trace
+from rasa_sdk.events import ActionExecuted, SlotSet
 
 from rasa_sdk.tracing.instrumentation import instrumentation
 from tests.tracing.instrumentation.conftest import MockActionExecutor
 from rasa_sdk.types import ActionCall
 from rasa_sdk import Tracker
 from rasa_sdk.tracing.tracer_register import ActionExecutorTracerRegister
+from rasa_sdk.executor import CollectingDispatcher
+
+
+dispatcher1 = CollectingDispatcher()
+dispatcher1.utter_message(template="utter_greet")
+dispatcher2 = CollectingDispatcher()
+dispatcher2.utter_message("Hello")
+dispatcher3 = CollectingDispatcher()
+dispatcher3.utter_message("Hello")
+dispatcher3.utter_message(template="utter_greet")
```

Comment:
These are disconnected from the test case, I'd try the following:
- place every dispatcher in a separate function and return the dispatcher
- parametrize the callable you want to use in the new unit test you create, instead of `messages` param
- call this parametrized function in the test and retrieve the dispatcher